app/models/imports/podcast_rss_import.rb

```diff
@@ -0,0 +1,293 @@
+require "feedjira"
+
+class PodcastRssImport < PodcastImport
+  store :config, accessors: [:episodes_only, :audio, :feed_rss], coder: JSON
+
+  has_many :episode_imports, dependent: :destroy, class_name: "EpisodeRssImport", foreign_key: :podcast_import_id
+
+  validate :validate_rss
+  validates :url, presence: true, http_url: true
+
+  def set_defaults
+    super
+    self.episodes_only ||= false
+    self.audio ||= {}
+  end
+
+  def feed_rss
+    config[:feed_rss] ||= http_get(url)
+  end
+
+  def feed
+    @feed ||= begin
+      parsed = Feedjira.parse(feed_rss)
+      unless parsed.is_a?(Feedjira::Parser::Podcast)
+        parser = begin
+          parsed.class.name.demodulize.underscore.humanize
+        rescue
+          ""
+        end
+        raise "Failed to validate #{url}, not a podcast feed: #{parser}"
+      end
+      parsed
+    end
+  end
+
+  def url=(value)
+    @feed = nil
+    self.feed_rss = nil
+    self.feed_episode_count = nil
+    super
+  end
+
+  def feed_rss=(value)
+    @feed = nil
+    super
+  end
+
+  def validate_rss
+    if url.blank?
+      errors.add(:url, :blank)
+    elsif !HttpUrlValidator.http_url?(url)
+      errors.add(:url, :not_http_url)
+    else
+      self.feed_episode_count = feed.entries.count
+    end
+  rescue ImportUtils::HttpError
+    errors.add(:url, :bad_http_response, message: "bad http response")
+  rescue
+    errors.add(:url, :invalid_rss, message: "invalid rss")
+  end
+
+  def import_metadata
+    !episodes_only
+  end
+
+  def import_metadata=(val)
+    self.episodes_only = (val == "0") ? true : !val
+  end
+
+  def config_url=(config_url)
+    self.config = JSON.parse(http_get(config_url))
+  end
+
+  def import!
+    status_started!
+
+    create_or_update_podcast!
+    create_or_update_episode_imports!
+
+    status_importing!
+  rescue => err
+    status_error!
+    unlock_podcast!
+    raise err
+  end
```

Comment:
Similarly, everything below here was copied over.

Changes above this:

1. `feed_rss` (the raw RSS string) is now saved in the `config` field as well.
2. `feed` (the parsed RSS) is memoized (provided some `feed_rss=` and `url=` setters to refresh it)
3. RSS parsing is now a custom validation `validate_rss`
4. `import_metadata` getter/setter to power the form checkbox (which is the inverse of the `episodes_only` db field).